misk/src/test/kotlin/misk/web/actions/SupportedHttpMethodsTest.kt

```diff
@@ -0,0 +1,103 @@
+package misk.web.actions
+
+import misk.inject.KAbstractModule
+import misk.testing.MiskTest
+import misk.testing.MiskTestModule
+import misk.web.Delete
+import misk.web.Get
+import misk.web.Post
+import misk.web.RequestBody
+import misk.web.WebActionModule
+import misk.web.WebTestingModule
+import misk.web.jetty.JettyService
+import misk.web.mediatype.MediaTypes
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import java.util.UUID
+import javax.inject.Inject
+
+// Tests all supported HTTP methods.
```

Comment:
Couldn't seem to find a similar test to add the DELETE test to. No strong reason to test GET and POST (since there's a bunch of other code already testing it), just erring on the side of having a test for all HTTP methods.

Happy to drop GET/POST tests if y'all feel it's redundant.

Reply:
I think the most useful thing to test here is whether a body is required or expected. Could test that GET declarations can't have a body, and we should decide whether we permit/require bodies for delete.

Reply:
Can do. The spec doesn't explicitly forbid message body for DELETE requests (okhttp also seems to accept one in the `RequestBuilder`, though it defaults to empty), but the [section on Message Body](https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.3) has something on ignoring them when there's no defined semantics:

> A server SHOULD read and forward a message-body on any request; if the request method does not include defined semantics for an entity-body, then the message-body SHOULD be ignored when handling the request.

I recall coming across libraries that don't even allow specifying a body so I don't mind being opinionated about disallowing body for DELETEs. How do you reckon it's the best way to disallow content? Validations on `WebActionServlet#doDelete`, prior to delegating to `#handleCall`?

Reply:
I think probably we should just forbid DELETE actions from declaring a request body. That way we blame the programmer for making a mistake, and don’t worry too much about what crazy internet users send us.

Maybe we do that in WebActionBinding.newWebActionBinder(). That can be a follow-up PR!